loopy/target/c/__init__.py

```diff
@@ -841,8 +841,11 @@ def get_function_declaration(
             # subkernel launches occur only as part of entrypoint kernels for now
             from loopy.schedule.tools import get_subkernel_arg_info
             skai = get_subkernel_arg_info(kernel, subkernel_name)
+            if (not self.target.single_subkernel_is_entrypoint
```

Comment:
Actually, I thought about this again. I think an error might not be the right thing, maybe you should just special case for `has_single_subkernel` or something similar in line 848. I also think that you have math callables like abs etc in the callables table so the check I proposed here is probably insufficient. Imho, I don't understand your code well enough. @kaushikcfd what do you think about this?

Reply:
I think this needed a minor edit. Proposed one in https://github.com/inducer/loopy/pull/701#discussion_r1019573280.